src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -181,36 +128,73 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
                     }
                     workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
 
-                    var useWorkloadSets = ShouldUseWorkloadSetMode(_sdkFeatureBand, _dotnetPath);
-                    useRollback = !string.IsNullOrWhiteSpace(_fromRollbackDefinition);
-
-                    _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(includePreviews, useWorkloadSets, offlineCache).Wait();
-
-                    if (useWorkloadSets)
+                    if (!_skipManifestUpdate)
                     {
-                        manifestsToUpdate = InstallWorkloadSet(context);
+                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) && File.Exists(installStateFilePath) && InstallStateContents.FromString(File.ReadAllText(installStateFilePath)).Manifests is not null)
+                        {
+                            //  If there is a rollback state file, then we don't want to automatically update workloads when a workload is installed
+                            //  To update to a new version, the user would need to run "dotnet workload update"
+                            //  TODO: We should also skip the update if a workload set version is pinned in install state or in global.json
+                            _skipManifestUpdate = true;
+                        }
                     }
-                    else
+
+                    RunInNewTransaction(context =>
                     {
-                        manifestsToUpdate = useRollback ? _workloadManifestUpdater.CalculateManifestRollbacks(_fromRollbackDefinition) :
-                            _workloadManifestUpdater.CalculateManifestUpdates().Select(m => m.ManifestUpdate);
-                    }
+                        if (_skipManifestUpdate)
+                        {
+                            _workloadInstaller.InstallWorkloads(workloadIds, _sdkFeatureBand, context, offlineCache);
+                        }
+                        else
+                        {
+                            if (Verbosity != VerbosityOptions.quiet && Verbosity != VerbosityOptions.q)
+                            {
+                                //  TODO: Change this message to account for workload set wording
+                                Reporter.WriteLine(LocalizableStrings.CheckForUpdatedWorkloadManifests);
+                            }
+                            UpdateWorkloads(context, workloadIds, offlineCache);
+                        }
+
+                        //  Write workload installation records
+                        var recordRepo = _workloadInstaller.GetWorkloadInstallationRecordRepository();
```

Comment:
We don't need to do this part before we install more workloads? Wouldn't this just be empty?